### ISSUE TITLE
Paid implementation consulting

### ISSUE THREAD

**Issue by philkunz** [haraka/Haraka#3147]
Is there any form of paid implementation consulting available from the devs?
Would be cool since it is hard to find freelancers that have expert knowledge with Haraka.

**Comment by msimerson** [comment]
Yes, subject to our availability and interest. Describe what you want developed.

**Comment by philkunz** [comment]
Thank you :) I sent you a contact request on LinkedIn for DMs.
Our intention is basically to to write a downstream npm package that simplifies DKIM etc. when having access to Cloudflare or similar DNS api, to enable super easy signed email sending without much config, basically with the easiness like mailgun, but for free.

**Comment by msimerson** [comment]
> Thank you :) I sent you a contact request on LinkedIn for DMs.

Good thing you mentioned that, I log onto LinkedIn about once every 2 years.

> Our intention is basically to to write a downstream npm package that simplifies DKIM

I like this idea. Does a workflow like this sound about right?

1. Haraka prepares to send an outbound message.
2. If the DKIM private key already exists, it signs and sends
3. The DKIM private key didn't exist, so:
    a. generate DKIM public and private keys
    b. publish DKIM public key to DNS
    c. install DKIM private key for Haraka
4. DKIM sign the message and send it
